pkg/repository/v1/sqlcv1/rate_limits.sql

```diff
@@ -0,0 +1,171 @@
+-- name: UpsertRateLimit :one
+INSERT INTO "RateLimit" (
+    "tenantId",
+    "key",
+    "limitValue",
+    "value",
+    "window"
+) VALUES (
+    @tenantId::uuid,
+    @key::text,
+    sqlc.arg('limit')::int,
+    sqlc.arg('limit')::int,
+    COALESCE(sqlc.narg('window')::text, '1 minute')
+) ON CONFLICT ("tenantId", "key") DO UPDATE SET
+    "limitValue" = sqlc.arg('limit')::int,
+    "window" = COALESCE(sqlc.narg('window')::text, '1 minute'),
+    "value" = CASE WHEN EXCLUDED."limitValue" < "RateLimit"."value" THEN EXCLUDED."limitValue" ELSE "RateLimit"."value" END
+RETURNING *;
+
+-- name: UpsertRateLimitsBulk :exec
+WITH input_values AS (
+    SELECT
+        unnest(@keys::text[]) AS "key",
+        unnest(@limitValues::int[]) AS "limitValue",
+        unnest(@windows::text[]) AS "window"
+)
+INSERT INTO "RateLimit" (
+    "tenantId",
+    "key",
+    "limitValue",
+    "value",
+    "window"
+)
+SELECT
+    @tenantId::uuid,
+    iv."key",
+    iv."limitValue",
+    iv."limitValue",
+    iv."window"
+FROM
+    input_values iv
+ON CONFLICT ("tenantId", "key") DO UPDATE SET
+    "limitValue" = EXCLUDED."limitValue",
+    "window" = EXCLUDED."window",
+    "value" = CASE WHEN EXCLUDED."limitValue" < "RateLimit"."value" THEN EXCLUDED."limitValue" ELSE "RateLimit"."value" END;
+
+-- name: CountRateLimits :one
+WITH rate_limits AS (
+    SELECT
+        rl."key"
+    FROM
+        "RateLimit" rl
+    WHERE
+        rl."tenantId" = @tenantId::uuid
+        AND (
+            sqlc.narg('search')::text IS NULL OR
+            rl."key" like concat('%', sqlc.narg('search')::text, '%')
```

Comment:
qq - how does this perform? also, is `like` okay here or is `ilike` better?

Reply:
Also from the old API/engine